credentials/alts/alts_test.go

```diff
@@ -337,8 +339,15 @@ func TestFullHandshake(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	c := testgrpc.NewTestServiceClient(conn)
-	if _, err = c.UnaryCall(ctx, &testpb.SimpleRequest{}, grpc.WaitForReady(true)); err != nil {
-		t.Errorf("c.UnaryCall() failed: %v", err)
+	for ; ctx.Err() == nil; <-time.After(5 * time.Second) {
```

Comment:
I think this should be more like this:
- You have an overall deadline of 10s
- You retry every 10ms
- You make the RPCs without the `WaitForReady` call option
- I'm not convinced that `DeadlineExceeded` is something where you retry. It should ideally be something else. You probably will see the correct code now if you make the RPC without the `WaitForReady` call option.

```golang
	const defaultTestTimeout = 10 * time.Second
	const defaultTestShortTimeout = 10 * time.Millisecond
	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
	defer cancel()
	c := testgrpc.NewTestServiceClient(conn)
	// TODO: A comment explaining why we need to retry the RPC until it succeeds.
	for ; ctx.Err() == nil; <-time.After(defaultTestShortTimeout) {
		_, err = c.UnaryCall(ctx, &testpb.SimpleRequest{})
		if err == nil {
			break
		}
		if code := status.Code(err); code == <expected code when handshake fails> {
			continue
		}
		t.Fatalf("c.UnaryCall() failed with unexpected error: %v", err)
	}
	if ctx.Err() != nil {
		t.Fatalf("Timeout when waiting for UnaryCall RPC to succeed")
	}
```

What do you think?

Reply:
Good call, the `WaitForReady` option was masking an `Unavailable` error code, which makes more sense. And thanks  for the tips! Implemented as you suggested. :)